base/gmp.jl

```diff
@@ -458,7 +475,7 @@ powermod(x::Integer, p::Integer, m::BigInt) = powermod(big(x), big(p), m)
 
 function gcdx(a::BigInt, b::BigInt)
     if b == 0 # shortcut this to ensure consistent results with gcdx(a,b)
-        return a < 0 ? (-a,-one(BigInt),zero(BigInt)) : (a,one(BigInt),zero(BigInt))
+        return a < 0 ? (-a,-ONE,ZERO) : (a,ONE,ZERO)
```

Comment:
The topic in the link (aliasing between arguments and return values of functions) is slightly different from the problem here (aliasing between arguments and global objects), but still very connected, as it refers to the safety of mutating the return value of a function (and whether it is always safe (cf. e.g. "computing something always involves returning a new object" in that thread) or depends on the runtime value of the argument).
Again as mutating `BigInt`s is not part of the public API (even if of course some code will mutate them for efficiency, as the language can not yet enforce their technical immutability), I think my aliasing code is OK, but I agree this is debatable (in particular as the expected speed-up here is not tremendous and concerns only the case of `b==0`).